app/src/main/java/com/techholding/android/examples/ui/firebase/google/GoogleFragment.kt

```diff
@@ -0,0 +1,111 @@
+package com.techholding.android.examples.ui.firebase.google
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.FirebaseApp
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.techholding.android.examples.R
+import com.techholding.android.examples.databinding.FragmentGoogleAuthBinding
+
+
+class GoogleFragment : Fragment() {
+
+    private var _binding: FragmentGoogleAuthBinding? = null
+    private val binding get() = _binding!!
+    private val googleViewModel: GoogleViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentGoogleAuthBinding.inflate(inflater, container, false)
+
+        initSignInButton()
+        initSignOutButton()
+
+        return binding.root
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        FirebaseApp.initializeApp(requireContext())
```

Comment:
This needs to be done only once during app startup so better to move in the Application class.